### ISSUE TITLE
mpt_basic assert

### ISSUE THREAD

**Issue by martinbremmer** [eclipse-cyclonedds/cyclonedds#343]
When running mpt_basic, it asserts:
```
=====================================================
Suite: helloworld
Test:  domain_42
=====================================================
--- [Subscriber(24147)] Start(-1) ...
=== [Publisher(24148)] Start(42) ...
--- [Subscriber(24147)] Waiting for 1 sample(s) ...
=== [Publisher(24148)] Send: { 24148, Now domain 42 is used }
--- [Subscriber(24147)] Received: { 24148, Now domain 42 is used }
mpt_basic: /home/martinb/cyclone/cyclonedds/src/core/ddsi/src/q_lease.c:75: void lease_management_term(struct q_globals *): Assertion `ddsrt_fibheap_min (&lease_fhdef, &gv->leaseheap) == NULL' failed.
mpt_basic: /home/martinb/cyclone/cyclonedds/src/core/ddsi/src/q_lease.c:75: void lease_management_term(struct q_globals *): Assertion `ddsrt_fibheap_min (&lease_fhdef, &gv->leaseheap) == NULL' failed.
Process helloworld::domain_42::pub failed (1)
Process helloworld::domain_42::sub kill(24147)
```

The backtrace is
```
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007f5024996801 in __GI_abort () at abort.c:79
#2  0x00007f502498639a in __assert_fail_base (fmt=0x7f5024b0d7d8 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", assertion=assertion@entry=0x7f5025c261a0 <.str> "ddsrt_fibheap_min (&lease_fhdef, &gv->leaseheap) == NULL", 
    file=file@entry=0x7f5025c26200 <.str.1> "/home/martinb/cyclone/cyclonedds/src/core/ddsi/src/q_lease.c", line=line@entry=75, 
    function=function@entry=0x7f5025c26260 <__PRETTY_FUNCTION__.lease_management_term> "void lease_management_term(struct q_globals *)") at assert.c:92
#3  0x00007f5024986412 in __GI___assert_fail (assertion=0x7f5025c261a0 <.str> "ddsrt_fibheap_min (&lease_fhdef, &gv->leaseheap) == NULL", file=0x7f5025c26200 <.str.1> "/home/martinb/cyclone/cyclonedds/src/core/ddsi/src/q_lease.c", 
    line=75, function=0x7f5025c26260 <__PRETTY_FUNCTION__.lease_management_term> "void lease_management_term(struct q_globals *)") at assert.c:101
#4  0x00007f5025a7f9d1 in lease_management_term (gv=0x6280000003a8) at /home/martinb/cyclone/cyclonedds/src/core/ddsi/src/q_lease.c:75
#5  0x00007f5025a7d679 in rtps_fini (gv=0x6280000003a8) at /home/martinb/cyclone/cyclonedds/src/core/ddsi/src/q_init.c:1715
#6  0x00007f5025b79de4 in dds_domain_free (vdomain=0x628000000100) at /home/martinb/cyclone/cyclonedds/src/core/ddsc/src/dds_domain.c:280
#7  0x00007f5025b85e73 in dds_entity_deriver_delete (e=0x628000000100) at /home/martinb/cyclone/cyclonedds/src/core/ddsc/src/dds__types.h:188
#8  0x00007f5025b873cf in really_delete_pinned_closed_locked (e=0x628000000100, delstate=DIS_IMPLICIT) at /home/martinb/cyclone/cyclonedds/src/core/ddsc/src/dds_entity.c:430
#9  0x00007f5025b87007 in dds_delete_impl_pinned (e=0x628000000100, delstate=DIS_IMPLICIT) at /home/martinb/cyclone/cyclonedds/src/core/ddsc/src/dds_entity.c:334
#10 0x00007f5025b8747b in really_delete_pinned_closed_locked (e=0x616000003380, delstate=DIS_EXPLICIT) at /home/martinb/cyclone/cyclonedds/src/core/ddsc/src/dds_entity.c:454
#11 0x00007f5025b87007 in dds_delete_impl_pinned (e=0x616000003380, delstate=DIS_EXPLICIT) at /home/martinb/cyclone/cyclonedds/src/core/ddsc/src/dds_entity.c:334
#12 0x00007f5025b86508 in dds_delete_impl (entity=471098270, delstate=DIS_EXPLICIT) at /home/martinb/cyclone/cyclonedds/src/core/ddsc/src/dds_entity.c:314
#13 0x00007f5025b86f05 in dds_delete (entity=471098270) at /home/martinb/cyclone/cyclonedds/src/core/ddsc/src/dds_entity.c:297
#14 0x0000000000517484 in MPT_ProcessEntry__hello_publisher (mpt__args__=0x7ffe9a6f8cd0, mpt__retval__=0x7ffe9a6f8cc0, domainid=42, topic_name=0x536840 <.str.10> "hello_42", sub_cnt=1, text=0x536880 <.str.11> "Now domain 42 is used")
    at /home/martinb/cyclone/cyclonedds/src/mpt/tests/basic/procs/hello.c:140
#15 0x0000000000519e37 in MPT_TestProcess__helloworld_domain_42_pub (mpt__args__=0x7ffe9a6f8cd0, mpt__retval__=0x7ffe9a6f8cc0) at /home/martinb/cyclone/cyclonedds/src/mpt/tests/basic/helloworld.c:56
#16 0x0000000000516797 in mpt_run_proc (proc=0x6030000001f0) at /home/martinb/cyclone/cyclonedds/build/src/mpt/tests/basic/mpt_basic.c:435
#17 0x0000000000514c2e in mpt_run_procs (spattern=0x7ffe9a6fa30e "helloworld", tpattern=0x7ffe9a6fa31c "domain_42", ppattern=0x7ffe9a6fa329 "pub") at /home/martinb/cyclone/cyclonedds/build/src/mpt/tests/basic/mpt_basic.c:452
#18 0x0000000000513f4d in main (argc=7, argv=0x7ffe9a6f8fb8) at /home/martinb/cyclone/cyclonedds/build/src/mpt/tests/basic/mpt_basic.c:594
```

Seems to have been introduced by #342.

**Comment by martinbremmer** [comment]
I reproduced it with clang and gcc (ubuntu 18.04). Also saw it on different tests.
But now I also have a few test runs without any problems. That suggests timing...

**Comment by dennis-adlink** [comment]
This seems to be related to participant dependency (participants on a node depend on a single DDSI participant for e.g. discovery and their lease). This is used by Opensplice: the Cyclone tests fail when OpenSplice is running and OpenSplice participants are discovered. I will look into this issue and create a fix.

**Comment by martinbremmer** [comment]
Fixed by #346.
